UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/TensorApplier.cs

```diff
@@ -0,0 +1,154 @@
+﻿using UnityEngine.MachineLearning.InferenceEngine;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.MachineLearning.InferenceEngine.Util;
+
+namespace MLAgents.InferenceBrain
+{
+    /// <summary>
+    /// A tensor Applier's Execute method takes a Tensor and a Dictionary of Agent to AgentInfo.
+    /// Uses the data contained inside the Tensor to modify the state of the Agent. The Tensors
+    /// are assumed to have the batch size on the first dimension and the agents to be ordered
+    /// the same way in the dictionary and in the Tensor.
+    /// </summary>
+    public interface TensorApplier
+    {
+        void Execute(Tensor tensor, Dictionary<Agent, AgentInfo> agentInfo);
+    }
+
+    /// <summary>
+    /// The Applier for the Continuous Action output tensor. Tensor is assumed to contain the
+    /// continuous action data of the agents in the batch.
+    /// </summary>
+    public class ContinuousActionOutputApplier : TensorApplier
+    {
+        public void Execute(Tensor tensor, Dictionary<Agent, AgentInfo> agentInfo)
+        {
+            var tensorDataAction = tensor.Data as float[,];
+            var actionSize = tensor.Shape[1];
+            var agentIndex = 0;
+            foreach (var agent in agentInfo.Keys)
+            {
+                var action = new float[actionSize];
+                for (var j = 0; j < actionSize; j++)
+                {
+                    action[j] = tensorDataAction[agentIndex, j];
+                }
+                agent.UpdateVectorAction(action);
+                agentIndex++;
+            }
+        }
+    }
+
+    /// <summary>
+    /// The Applier for the Discrete Action output tensor. Uses multinomial to sample discrete
+    /// actions from the logits contained in the tensor.
+    /// </summary>
+    public class DiscreteActionOutputApplier : TensorApplier
+    {
+        private int[] _actionSize;
+        private Multinomial _multinomial;
+        
+        public DiscreteActionOutputApplier(int[] actionSize, int seed)
```

Comment:
Nice! A lot more streamlined than passing params to static function

Reply:
Won't fix.